sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -653,6 +653,19 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with Logging {
   }
 
   /**
+   * Create a table-valued function call with arguments, e.g. range(1000)
+   */
+  override def visitTableValuedFunction(ctx: TableValuedFunctionContext)
+      : LogicalPlan = withOrigin(ctx) {
+    val expressions = ctx.expression.asScala.map { ec =>
+      val e = expression(ec)
+      assert(e.foldable, "All arguments of a table-valued-function must be constants.", ec)
```

Comment:
Yeah, lets defer evaluation until you need them in `ResolveTableValuedFunctions`. You will need to add a check though.